### ISSUE TITLE
Deleting VMSS does not continue if previous long running operation fails

### ISSUE THREAD

**Issue by mweibel** [kubernetes-sigs/cluster-api-provider-azure#2671]
/kind bug

**What steps did you take and what happened:**
1. have a machinepool which can not join a cluster (bootstrap token expired or simply override kubeadm join command with something invalid)
2. scale up machinepool
3. notice instance can not join the cluster and fails with bootstrapping error
4. try to delete machinepool

At this point the AzureMachinePool still has the LongRunningOperationState set to the previous long running operation:
```
  longRunningOperationStates:
  - data: eyBASE64Encoded
    name: a-vmss-name
    resourceGroup: a-resource-group-name
    serviceName: scalesets
    type: PATCH
```
And the following error appears in the log:

```
capz-controller-manager-96c6dd5fb-z5rn7 manager E0927 12:15:35.426801      12 controller.go:326]  "msg"="Reconciler error" "error"="error deleting AzureMachinePool anamespace/a-vmss-name: failed to delete AzureMachinePool service scalesets: failed to get result from future: failed checking if the operation was complete: Code=\"VMExtensionProvisioningError\" Message=\"VM has reported a failure when processing extension 'CAPZ.Windows.Bootstrapping'. Error message: \\\"Command execution finished, but failed because it returned a non-zero exit code of: '-2'\\\"\\r\\n\\r\\nMore information on troubleshooting is available at https://aka.ms/vmextensionwindowstroubleshoot \"" "azureMachinePool"={"name":"a-vmss-name","namespace":"a-namespace"} "controller"="azuremachinepool" "controllerGroup"="infrastructure.cluster.x-k8s.io" "controllerKind"="AzureMachinePool" "name"="a-vmss-name" "namespace"="a-namespace" "reconcileID"="7597183e-b5a2-4513-8132-00a4acb56c7b"
```

Looking at the code:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/scalesets/scalesets.go#L185-L191

It seems to me that if any previous long running operation is set and failed, it's not removing it from the state and therefore is unable to delete the AzureMachinePool. 

**What did you expect to happen:**
Deletion is possible. 

**Environment:**

- cluster-api-provider-azure version: v1.5.0
- Kubernetes version: (use `kubectl version`): 1.24
- OS (e.g. from `/etc/os-release`): linux/windows hybrid

**Comment by CecileRobertMichon** [comment]
This should get fixed by #2665 

cc @nojnhuh

**Comment by mweibel** [comment]
true. I didn't validate it but reading the code this should be fixed by #2665. 
Thanks. Closing.
